app/code/Magento/Sales/Model/Order/Email/SenderBuilder.php

```diff
@@ -37,25 +38,41 @@ class SenderBuilder
      * @param IdentityInterface $identityContainer
      * @param TransportBuilder $transportBuilder
      * @param TransportBuilderByStore $transportBuilderByStore
+     * @param TransportBuilderFactory $transportBuilderFactory
      */
     public function __construct(
         Template $templateContainer,
         IdentityInterface $identityContainer,
         TransportBuilder $transportBuilder,
-        TransportBuilderByStore $transportBuilderByStore = null
+        TransportBuilderByStore $transportBuilderByStore = null,
+        TransportBuilderFactory $transportBuilderFactory = null
     ) {
         $this->templateContainer = $templateContainer;
         $this->identityContainer = $identityContainer;
         $this->transportBuilder = $transportBuilder;
+        $this->transportBuilderFactory = $transportBuilderFactory ? $transportBuilderFactory : \Magento\Framework\App\ObjectManager::getInstance()->get(\Magento\Framework\Mail\Template\TransportBuilderFactory::class);
     }
 
+    /**
+     * Create a new transport builder to stop data persisting between customers
+     *
+     * @return void
+     */
+    protected function resetTransportBuilder()
```

Comment:
Please make methods private whenever possible.

Reply:
Done

Reply:
The method still protected, please, make it private